nixos/modules/hardware/video/nvidia.nix

```diff
@@ -343,14 +343,25 @@ in
             ];
 
             # Don't add `nvidia-uvm` to `kernelModules`, because we want
-            # `nvidia-uvm` be loaded only after `udev` rules for `nvidia` kernel
-            # module are applied.
+            # `nvidia-uvm` be loaded only after the GPU device is available, i.e. after `udev` rules
+            # for `nvidia` kernel module are applied.
+            # This matters on Azure GPU instances: https://github.com/NixOS/nixpkgs/pull/267335
             #
             # Instead, we use `softdep` to lazily load `nvidia-uvm` kernel module
             # after `nvidia` kernel module is loaded and `udev` rules are applied.
             extraModprobeConfig = ''
               softdep nvidia post: nvidia-uvm
             '';
+
+            # Exception is the open-source kernel module failing to load nvidia-uvm using softdep
+            # for unknown reasons.
+            # It affects CUDA: https://github.com/NixOS/nixpkgs/issues/334180
+            # Previously nvidia-uvm was explicitly loaded only when xserver was enabled:
+            # https://github.com/NixOS/nixpkgs/pull/334340/commits/4548c392862115359e50860bcf658cfa8715bde9
+            # We are now loading the module eagerly for all users because it affects headless systems too.
```

Comment:
- all users - users of the open driver
- "because it affects headless systems too" - is tautological; we started with the conservative conditional because the module was already conflating the meaning of `xserver.enable` and `videoDrivers`; we can't easily form a conditional for "all and only systems where softdep fails", the closest we could do is inspect `hardware.nvidia.package` but I think that's too implicit